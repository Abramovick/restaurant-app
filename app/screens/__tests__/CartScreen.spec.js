import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { shallow, } from 'enzyme';
import ConnectedCartScreen, { CartScreen } from '../CartScreen';

// Mocked component
jest.mock('../CheckOutScreen', () => 'CheckOut');
jest.mock('redux-form/lib/Field', () => 'Field');


// Test objects
const items = [{
  name: 'mariam',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world '
},
{
  name: 'tope',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world'
}];
const emptyItem = [];

const navigation = {
  navigate: jest.fn(),
  state: {
    params: {
      name: 'mariam',
      ImageUrl: 'gdfgds',
      Price: 700,
      description: 'the hello is in the world '
    }
  }
};

// Mocked functions
const decreaseItemQuantity = jest.fn();
const increaseItemQuantity = jest.fn();

// // Test suites
describe('CARTSCREEN --- Snapshot', () => {
  it('renders the cart screen if user is logged in', async () => {
    const tree = renderer.create(<CartScreen
      items={items}
      navigation={navigation}
      increaseItemQuantity={increaseItemQuantity}
      decreaseItemQuantity={decreaseItemQuantity}
      />);
    expect(tree.getInstance().redirectToShop()).toMatchSnapshot();
    expect(tree.getInstance().closeModal()).toMatchSnapshot();
    expect(tree.getInstance().openModal()).toMatchSnapshot();
    expect(tree).toMatchSnapshot();
  });


  it('renders a logout/login view is user is not logged in', async () => {
    const tree = renderer.create(<CartScreen
      items={emptyItem}
      navigation={navigation}
      increaseItemQuantity={increaseItemQuantity}
      decreaseItemQuantity={decreaseItemQuantity}
      />);

    expect(tree).toMatchSnapshot();
  });
});

describe('CHECKOUTSCREEN --- Shallow rendering + passing the store directly', () => {
  const initialState = {
    cart: {
      items
    },
    total: 453543
  };
  const mockStore = configureMockStore();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedCartScreen store={store}/>);
  });

  it('renders the cart component', () => {
    container = shallow(<CartScreen
      items={items}
      increaseItemQuantity={increaseItemQuantity}
      decreaseItemQuantity={decreaseItemQuantity}
    />);
    container.find('Cart').dive().find('TouchableHighlight').first()
      .simulate('press');
    container.find('Cart').dive().find('TouchableHighlight').at(1)
      .simulate('press');
    expect(container.length).toEqual(1);
  });

  it('renders the connected component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('items'))
      .toEqual(initialState.cart.items);
    expect(container.prop('total'))
      .toEqual(initialState.cart.total);
  });
});
