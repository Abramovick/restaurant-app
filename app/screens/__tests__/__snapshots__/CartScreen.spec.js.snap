// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARTSCREEN --- Snapshot renders a logout/login view is user is not logged in 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 7,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "padding": 20,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              },
              Object {
                "alignSelf": "center",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 5,
                "paddingTop": 0,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={15}
            smallFont={true}
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "fontSize": 15,
                },
              ]
            }
          >
             SORRY, YOUR CART IS EMPTY
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              },
              Object {
                "alignSelf": "center",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 5,
                "paddingTop": 0,
              },
            ]
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 28,
                    "width": 35,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            ïˆ†
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              },
              Object {
                "alignSelf": "center",
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 5,
                "paddingTop": 0,
              },
            ]
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#F7C04C",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 2,
                "flexDirection": "row",
                "height": 35,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              smallFont={true}
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 11,
                  },
                ]
              }
            >
                  GO TO SHOP    
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`CARTSCREEN --- Snapshot renders the cart screen if user is logged in 1`] = `undefined`;

exports[`CARTSCREEN --- Snapshot renders the cart screen if user is logged in 2`] = `undefined`;

exports[`CARTSCREEN --- Snapshot renders the cart screen if user is logged in 3`] = `undefined`;

exports[`CARTSCREEN --- Snapshot renders the cart screen if user is logged in 4`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 7,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={undefined}
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "padding": 7,
      }
    }
  >
    <View>
      <View
        style={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderRightColor": "#B2BEB5",
                    "borderRightWidth": 0.5,
                    "padding": 6,
                    "width": "40%",
                  },
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                    },
                  ]
                }
              >
                ITEM
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRightColor": "#B2BEB5",
                    "borderRightWidth": 0.5,
                    "padding": 6,
                    "width": "40%",
                  },
                  Object {
                    "width": "15%",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                    },
                  ]
                }
              >
                No
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRightColor": "#B2BEB5",
                    "borderRightWidth": 0.5,
                    "padding": 6,
                    "width": "40%",
                  },
                  Object {
                    "width": "15%",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                    },
                  ]
                }
              >
                Price
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRightColor": "#B2BEB5",
                    "borderRightWidth": 0.5,
                    "padding": 6,
                    "width": "40%",
                  },
                  Object {
                    "borderRightWidth": 0,
                    "height": "100%",
                    "padding": 6,
                    "width": "20%",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                    },
                  ]
                }
              >
                SubTotal
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "7%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "row",
                  "padding": 6,
                  "width": "40%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "height": 40,
                    "marginTop": 5,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginLeft={6}
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginLeft": 6,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                mariam
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "height": "100%",
                  "padding": 6,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "height": "100%",
                  "padding": 6,
                  "width": "15%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                â‚¦
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRightWidth": 0,
                  "height": "100%",
                  "padding": 6,
                  "width": "20%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                â‚¦
                NaN
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "7%",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "flex": 1,
                  "flexDirection": "row",
                  "padding": 6,
                  "width": "40%",
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "height": 40,
                    "marginTop": 5,
                    "width": 40,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginLeft={6}
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginLeft": 6,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                tope
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "height": "100%",
                  "padding": 6,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "borderRightColor": "#f5f5f5",
                  "borderRightWidth": 0.5,
                  "height": "100%",
                  "padding": 6,
                  "width": "15%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                â‚¦
              </Text>
            </View>
            <View
              style={
                Object {
                  "borderRightWidth": 0,
                  "height": "100%",
                  "padding": 6,
                  "width": "20%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                marginTop={15}
                smallFont={true}
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "fontSize": 11,
                      "marginTop": 15,
                    },
                  ]
                }
              >
                â‚¦
                NaN
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "7%",
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "wrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "alignSelf": "center",
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingTop": 0,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "center",
                  "flex": 0.8,
                }
              }
              virtual={undefined}
            >
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  fontSize={11}
                  smallFont={true}
                  style={
                    Array [
                      Object {
                        "fontFamily": "SinkinSans-200XLight",
                        "fontSize": 12,
                      },
                      Object {
                        "fontSize": 11,
                      },
                    ]
                  }
                >
                  TOTAL:       â‚¦
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            smallFont={true}
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "fontSize": 11,
                },
              ]
            }
          >
                CHECKOUT   
          </Text>
        </View>
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onModalHide={[Function]}
          onRequestClose={[Function]}
          transparent={true}
          visible={true}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "black",
                "bottom": 0,
                "height": 1334,
                "left": 0,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 750,
              }
            }
            testID={undefined}
          />
          <View
            collapsable={undefined}
            onModalHide={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
                "margin": 37.5,
                "transform": Array [
                  Object {
                    "translateY": 1334,
                  },
                ],
              }
            }
          >
            <CheckOut
              closeModal={[Function]}
              navigation={
                Object {
                  "navigate": [Function],
                  "state": Object {
                    "params": Object {
                      "ImageUrl": "gdfgds",
                      "Price": 700,
                      "description": "the hello is in the world ",
                      "name": "mariam",
                    },
                  },
                }
              }
            />
          </View>
        </Modal>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
